tests/lib/prepare.sh

```diff
@@ -223,15 +223,29 @@ EOF
         cp /usr/bin/snap $IMAGE_HOME
         export UBUNTU_IMAGE_SNAP_CMD=$IMAGE_HOME/snap
 
-        # download pc-kernel snap for the specified channel
-        snap download --channel="$KERNEL_CHANNEL" pc-kernel
+        EXTRA_FUNDAMENTAL=
+        IMAGE_CHANNEL=edge
+        if [ "$KERNEL_CHANNEL" = "$GADGET_CHANNEL" ]; then
+            IMAGE_CHANNEL=$KERNEL_CHANNEL
+        # we assume that only one of KERNEL_CHANNEL and GADGET_CHANNEL will be different from edge
+        elif [ "$KERNEL_CHANNEL" != edge ]; then
+            # download pc-kernel snap for the specified channel
+            snap download --channel="$KERNEL_CHANNEL" pc-kernel
+
+            EXTRA_FUNDAMENTAL='--extra-snaps '"$PWD"'/pc-kernel_*.snap'
+        elif [ "$GADGET_CHANNEL" != edge ]; then
```

Comment:
Yes, this is meant to be used from spread-cron after detecting a new fundamental snap has been promoted to any of the channels, only one of them should be different from the default image channel (currently edge).